internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -269,15 +370,19 @@ func (u *Upgrader) sourceURI(retrievedURI string) string {
 	return u.settings.SourceURI
 }
 
-func rollbackInstall(ctx context.Context, log *logger.Logger, hash string) {
-	os.RemoveAll(filepath.Join(paths.Data(), fmt.Sprintf("%s-%s", agentName, hash)))
-	_ = ChangeSymlink(ctx, log, release.ShortCommit())
+func rollbackInstall(ctx context.Context, log *logger.Logger, topDirPath, versionedHome string) {
+	err := os.RemoveAll(filepath.Join(topDirPath, versionedHome))
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		// TODO should this be a warning or an error ?
+		log.Warnw("error rolling back install", "error.message", err)
```

Comment:
It looks like this is only failing to remove the version we are upgrading to.

If another upgrade attempt can succeed with this situation, it's a warning. If not, it's an error.

Also I think the log message text could be clearer to state it is an error removing a path along with which path that is.

Reply:
I changed it a bit in the latest commit, I think it should be ok now